code/controllers/subsystem/inactivity.dm

```diff
@@ -1,20 +1,25 @@
-#define INACTIVITY_KICK 6000 //10 minutes in ticks (approx.)
+#define INACTIVITY_KICK 10 MINUTES
 
 SUBSYSTEM_DEF(inactivity)
 	name = "Inactivity"
-	wait = INACTIVITY_KICK
+	wait = INACTIVITY_KICK * 0.5
 	flags = SS_NO_INIT | SS_BACKGROUND
 	priority = SS_PRIORITY_INACTIVITY
 	runlevels = RUNLEVELS_DEFAULT|RUNLEVEL_LOBBY
 
 /datum/controller/subsystem/inactivity/fire(resumed = FALSE)
-	if (CONFIG_GET(flag/kick_inactive))
-		for(var/i in GLOB.clients)
-			var/client/C = i
-			if(C.admin_holder && C.admin_holder.rights & R_ADMIN) //Skip admins.
-				continue
-			if (C.is_afk(INACTIVITY_KICK))
-				if (!istype(C.mob, /mob/dead))
-					log_access("AFK: [key_name(C)]")
-					to_chat(C, SPAN_WARNING("You have been inactive for more than 10 minutes and have been disconnected."))
-					qdel(C)
+	// Maybe we should just get a SS to bandaid all important global lists like this?
```

Comment:
that's what /tg/ does, they have a "maintenance" subsystem for that. it might be better to port it, or do one of our own, then we can run it much faster. null in client lists will break down various things, not just cause runtimes

Reply:
Ok - are there any other lists you know are problematic it should also clean up even if at a lower frequency?

Reply:
Since GLOB.clients is all we are concerned about for now, I will consider null clients to be inactive and they will be kicked from the list here ;)